From: Nicolas Michaut <nicolas@yumilab.com>
Subject: [PATCH] ARM: dts: sun8i-h3: add SmartPi One device tree
Date: 2026-05-27

Add device tree for the Yumi Lab SmartPi One board.
Same hardware as NanoPi M1 (Allwinner H3, 1GB RAM), with
custom model and compatible strings.

Signed-off-by: Nicolas Michaut <nicolas@yumilab.com>
---
 arch/arm/boot/dts/allwinner/Makefile              |   1 +
 arch/arm/boot/dts/allwinner/sun8i-h3-smartpi-one.dts | 100 ++++++++++++++++++
 2 files changed, 101 insertions(+)
 create mode 100644 arch/arm/boot/dts/allwinner/sun8i-h3-smartpi-one.dts

diff --git a/arch/arm/boot/dts/allwinner/Makefile b/arch/arm/boot/dts/allwinner/Makefile
--- a/arch/arm/boot/dts/allwinner/Makefile
+++ b/arch/arm/boot/dts/allwinner/Makefile
@@ -1,4 +1,5 @@
 # SPDX-License-Identifier: GPL-2.0
+dtb-$(CONFIG_MACH_SUN8I_H3) += sun8i-h3-smartpi-one.dtb

 dtb-$(CONFIG_MACH_SUN4I) += \
 	sun4i-a10-a1000.dtb \
diff --git a/arch/arm/boot/dts/allwinner/sun8i-h3-smartpi-one.dts b/arch/arm/boot/dts/allwinner/sun8i-h3-smartpi-one.dts
new file mode 100644
--- /dev/null
+++ b/arch/arm/boot/dts/allwinner/sun8i-h3-smartpi-one.dts
@@ -0,0 +1,100 @@
+// SPDX-License-Identifier: (GPL-2.0+ OR MIT)
+/*
+ * Copyright (C) 2026 Yumi Lab <nicolas@yumilab.com>
+ *
+ * Based on sun8i-h3-nanopi-m1.dts by Milo Kim
+ * Same hardware, custom branding for Yumi SmartPi One.
+ */
+
+#include "sun8i-h3-nanopi.dtsi"
+
+/ {
+	model = "YUMi SmartPi One";
+	compatible = "yumi,smartpi-one", "allwinner,sun8i-h3";
+
+	aliases {
+		ethernet0 = &emac;
+	};
+
+	connector {
+		compatible = "hdmi-connector";
+		type = "a";
+
+		port {
+			hdmi_con_in: endpoint {
+				remote-endpoint = <&hdmi_out_con>;
+			};
+		};
+	};
+};
+
+&de {
+	status = "okay";
+};
+
+&ehci1 {
+	status = "okay";
+};
+
+&ehci2 {
+	status = "okay";
+};
+
+&emac {
+	phy-handle = <&int_mii_phy>;
+	phy-mode = "mii";
+	allwinner,leds-active-low;
+	status = "okay";
+};
+
+&hdmi {
+	status = "okay";
+};
+
+&hdmi_out {
+	hdmi_out_con: endpoint {
+		remote-endpoint = <&hdmi_con_in>;
+	};
+};
+
+&ir {
+	pinctrl-names = "default";
+	pinctrl-0 = <&r_ir_rx_pin>;
+	status = "okay";
+};
+
+&ohci1 {
+	status = "okay";
+};
+
+&ohci2 {
+	status = "okay";
+};
--
2.43.0
